### PR TITLE
Normalize the loss value to account for the number of gradient accumulation steps

### PR DESCRIPTION
**Pull request by chrisociepa**
As discussed in #168 the loss value should be divided by gradient_accumulation_steps

### CODE REVIEW

**Comment by karpathy** [comment]
This seems too complicated? 

The PyTorch docs snippet here 
https://pytorch.org/docs/stable/notes/amp_examples.html#gradient-accumulation

just simply divides the loss by the number of grad accum steps, would that work?

**Comment by karpathy** [comment]
@chrisociepa I fixed the issue with https://github.com/karpathy/nanoGPT/commit/553f949f46e04a946e5ff2da6e4bb70a221ea1fc . Note that I am only approximating the true loss by multiplying by num gradient accum, but this suffices just to get approximate numbers printed in the console. wandb will have the correct values regardless. This also allows us to avoid very expensive `.item()` calls, which actually cause a GPU CPU sync point and can significantly slow down running times. Thanks!
